core/src/avm1.rs

```diff
@@ -1503,6 +1518,24 @@ impl<'gc> Avm1<'gc> {
         self.set_variable(context, var_path.as_string()?, value)
     }
 
+    #[allow(clippy::float_cmp)]
+    fn action_strict_equals(&mut self, _context: &mut ActionContext) -> Result<(), Error> {
+        // The same as normal equality but types must match
+        let a = self.pop()?;
+        let b = self.pop()?;
+        let result = match (b, a) {
+            (Value::Undefined, Value::Undefined) => true,
+            (Value::Null, Value::Null) => true,
+            (Value::Number(a), Value::Number(b)) => a == b || (a.is_nan() && b.is_nan()),
+            (Value::String(a), Value::String(b)) => a == b,
+            (Value::Bool(a), Value::Bool(b)) => a == b,
+            (Value::Object(_a), Value::Object(_b)) => false, // TODO
+            _ => false,
+        };
```

Comment:
```suggestion
let result = a == b;
```

`Value` implements `PartialEq` which, as far as I can tell, is an identity equality check. It even implements for Object.

Reply:
@Dinnerbone you're absolutely correct. Just checked and the 2 implementations are semantically the same. Thank you :)